modules/swagger-codegen/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -14,7 +14,12 @@ import { Response, ResponseContentType }                     from '@angular/http
 import { CustomQueryEncoderHelper }                          from '../encoder';
 {{/useHttpClient}}
 
+{{^useRxJS6}}
 import { Observable }                                        from 'rxjs/Observable';
+{{/useRxJS6}}
+{{#useRxJS6}}
+import { Observable }                                    from 'rxjs';
+{{/useRxJS6}}
 {{^useHttpClient}}
 import '../rxjs-operators';
```

Comment:
the `import '../rxjs-operators';` should be removed, as pipeable operators are already available as of RxJS 5.5

Reply:
Sorry, if I misunderstood, the rxjs-operators is imported in a {{^useHttpClient}} {{/useHttpClient}} - I (wrongly?) thought, this meant that these imports are only used if angular is older than 4.3, and that after 4.3 they are not imported anymore.

For RxJS 6.1, the operators would need to be imported differently, if they are used - see: https://github.com/ReactiveX/rxjs/blob/master/MIGRATION.md#howto-convert-to-pipe-syntax .

From what I can see, actually, of the operators, throw & catch are not used by the api service, only map is. However I just found out, that at least at one point map is used, which will need to be converted to a .pipe( map( ()=> {}), ) for rxjs 6. Let me update the PR and get back to you.

Reply:
Update: The map operator is also only generated when ```{{^useHttpClient}} {{/useHttpClient}}``` - therefore not used in the updated angular 6 code and should not need to be imported.

Reply:
ah, you are right, I missed that it is in a `{{^useHttpClient}}{{/useHttpClient}}` block.

you don't need to convert the `.map()` to `.pipe(map())`, since it is only used is here: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/typescript-angular/api.service.mustache#L89

And since this line is only used for Angular <4.3, it should be fine as it is

Reply:
My bad, I misread. You want to move it to the ```{{^useRxJS6}}{{/useRxJS6}}``` block, which makes sense. I'll update the PR.

Reply:
actually I think we shouldn't move it to the `{{^useRxJS6}}{{/useRxJS6}}` block, since this would also affect Angular `>=4.3` and `<6`

Reply:
After moving it, angular 5 test fails =(

Reply:
yes, that's what I thought would happen. so you can simply ignore the initial review comment that stated

> the import '../rxjs-operators'; should be moved inside the {{^useRxJS6}}{{/useRxJS6}} block, see my general review comment.